src/background/ui.js

```diff
@@ -408,14 +411,15 @@ export class UI extends Component {
       text = "badgeWarningText";
     }
 
-    return Promise.all([
-      browser.browserAction.setIcon({
-        path: icon,
-      }),
-      browser.browserAction.setTitle({
-        title: this.getTranslation(text),
-      }),
-    ]);
+    await browser.browserAction.setTitle({
```

Comment:
I would revert these changes and add: "if (constants.isAndroid) {\n  return;\n}\n" at the beginning of updateIcon().